CMakeLists.txt

```diff
@@ -43,6 +43,8 @@ if(NOT MSVC)
 else()
 	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -D_CRT_SECURE_NO_WARNINGS /wd4003 /wd4244 /wd4146 /wd4267 /Zc:preprocessor-")
 	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D_CRT_SECURE_NO_WARNINGS /wd4003 /wd4244 /wd4146 /wd4267 /Zc:preprocessor-")
+	#MSVC debug does not like both .exe and shared .dll to be named the same, as it prevents loading of debug symbols
+ set(BINARY_NAME mgbalib)
```

Comment:
Your indentation does not match the rest of the file. This is also not the only thing BINARY_NAME is  be used for, so you'll need to narrow it down with the actual library file.